Add endpoint for ekadashi date

Lighty0410 opened the pull request:
Closes:#7
Closes:#8

Lighty0410 committed:
Add endpoint, add function in mongo for endpoint, minor change in server

Lighty0410 committed:
Add commentary for SendEkadashi

Lighty0410 committed:
Merge remote-tracking branch 'origin/master' into ekadashi-endpoint

Lighty0410 committed:
Change function name from SendEkadashi to RetrieveEkadashi

Lighty0410 committed:
Merge branch 'ekadashi-logic' into ekadashi-endpoint

Lighty0410 committed:
Refactor ekadashi-endpoint and moon.go

Lighty0410 committed:
Replace import, delete staticcheck from Makefile

sashayakovtseva reviewed (changes requested):
Changes should be reflected in `openAPI.yaml`

Lighty0410 committed:
Replace /users endpoint to /ekadashi/next. Thus change schema type

Lighty0410 committed:
Merge remote-tracking branch 'origin/master' into ekadashi-endpoint

Lighty0410 committed:
Update server, delete ShowAllUsers from server

Lighty0410 committed:
Update function name and commentary

Lighty0410 committed:
Update function name in endpoint, delete unused ShowAllusers function

sashayakovtseva reviewed (changes requested):
No tests for LastEkadashi

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +22,25 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
+// LastEkadashi retrieves information about the last ekadashi date from the database.
+func (s *Service) LastEkadashi(day time.Time) (EkadashiDate, error) {
+	c := s.db.Collection("ekadashi")
+	cur, err := c.Find(context.Background(), bson.D{{
```
Can you use FindOne?

sashayakovtseva commented on pkg/server/ekadashi-endpoint.go:
```diff
@@ -0,0 +1,31 @@
+	"github.com/Lighty0410/ekadashi-server/pkg/mongo"
+)
+
+func (s *EkadashiServer) showEkadashiEnpoint(w http.ResponseWriter, r *http.Request) {
```
`nextEkadashiHandler`
also filename can be simplified to simply `ekadashi`

Lighty0410 committed:
Refactor LastEkadashi, change function name/filename

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +22,18 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
+		}},
+	}}).Decode(ekadashiDay)
+	if err != nil {
+		return ekadashiDay, fmt.Errorf("cannot find user: %v", err)
```
cannot find last ekadashi

sashayakovtseva commented on pkg/server/ekadashi.go:
```diff
@@ -0,0 +1,31 @@
+		jsonError(w, http.StatusInternalServerError, fmt.Errorf("cannot check authentification: %v", err))
+		return
+	}
+	ekadashiDate, err := s.db.LastEkadashi(time.Now())
```
You need next ekadashi here, not last

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +22,18 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
 }
+
+// LastEkadashi retrieves information about the last ekadashi date from the database.
+func (s *Service) LastEkadashi(day time.Time) (EkadashiDate, error) {
```
Still missing unit tests

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +22,18 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
+	c := s.db.Collection("ekadashi")
+	err := c.FindOne(context.Background(), bson.D{{
+		Key: "date", Value: bson.D{{
+			Key: "$gt", Value: day,
```
There is no guarantees for documents order, you need to add sort option

Lighty0410 committed:
Add tests, add additional option for FindOne, minor refactor

sashayakovtseva commented on pkg/mongo/database_test.go:
```diff
@@ -57,25 +60,101 @@ func TestAddAndReadUser(t *testing.T) {
+
+func TestService_NextEkadashiAndAddEkadashi(t *testing.T) {
+	connectionURL := "mongodb://localhost:27017"
+	/*if connectionURL == "" {
```
REMOVE DEBUG LINES!!!!!!

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +24,23 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
+
+// NextEkadashi retrieves information about the last ekadashi date from the database.
+func (s *Service) NextEkadashi(day time.Time) (EkadashiDate, error) {
+	if day.IsZero() {
```
WHY?

sashayakovtseva commented on pkg/mongo/moon.go:
```diff
@@ -20,3 +24,23 @@ func (s *Service) AddEkadashi(day *EkadashiDate) error {
+	c := s.db.Collection("ekadashi")
+	err := c.FindOne(context.Background(), bson.D{{
+		Key: "date", Value: bson.D{{
+			Key: "$gt", Value: day.AddDate(0, 0, -1),
```
why?

sashayakovtseva commented on pkg/server/ekadashi.go:
```diff
@@ -0,0 +1,31 @@
+	}
+	ekadashiDate, err := s.db.NextEkadashi(time.Now())
+	if err != nil {
+		jsonError(w, http.StatusInternalServerError, fmt.Errorf("there is no information in database: %v", err))
```
return!

sashayakovtseva replied:
could not find next ekadashi

Lighty0410 committed:
Refactor code

1. Add more informative tests. 
2. Change "jsonError" commentary , add return in handler. 
3.  Delete unnecessary check in NextEkadashi

Lighty0410 committed:
Fix env var name in tests
